hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java

```diff
@@ -0,0 +1,693 @@
+                                HTS_CALLS,
+                                "burnTokenCall",
+                                HapiParserUtil.asHeadlongAddress(
+                                        asAddress(spec.registry().getTokenID(FUNGIBLE_TOKEN))),
+                                BigInteger.valueOf(amountToBurn),
+                                new long[0])
+                                .via(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
+                                .gas(GAS_TO_OFFER)
+                                .payingWith(SIGNER)
+                                .signedBy(SIGNER),
+                        // verify that the total supply of the tokens is not affected
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount),
+                        getTxnRecord(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
+                                .andAllChildRecords()
+                                .logged(),
+                        // Create a key with thresh 1/2 with sigs:  new ed25519 key, contractId of MINT_CONTRACT
+                        // contract. MINT_CONTRACT is used only as a "wrong" contract id
+                        newKeyNamed(TRESHOLD_KEY_WITH_SIGNER_KEY)
+                                .shape(TRESHOLD_KEY_SHAPE.signedWith(sigs(ON, MINT_CONTRACT))),
+                        // Update the signer of the transaction to have the threshold key with the wrong contract id
+                        cryptoUpdate(SIGNER).key(TRESHOLD_KEY_WITH_SIGNER_KEY),
+                        // Update the token's supply to have the threshold key with the wrong contract id
+                        tokenUpdate(FUNGIBLE_TOKEN).supplyKey(TRESHOLD_KEY_WITH_SIGNER_KEY),
+                        // Test Case 2: Signer paying and signing a token burn transaction, when the token
+                        // is expected to  be burned by the token treasury
+                        // SIGNER → call → CONTRACT → call → PRECOMPILE
+                        // The signer and the token have a threshold key with the signer's public key
+                        // and the wrong contract id (MINT_CONTRACT)
+                        contractCall(
+                                HTS_CALLS,
+                                "burnTokenCall",
+                                HapiParserUtil.asHeadlongAddress(
+                                        asAddress(spec.registry().getTokenID(FUNGIBLE_TOKEN))),
+                                BigInteger.valueOf(amountToBurn),
+                                new long[0])
+                                .via(SIGNER_MINTS_WITH_SIGNER_PUBLIC_KEY_AND_WRONG_CONTRACT_ID)
+                                .gas(GAS_TO_OFFER)
+                                .alsoSigningWithFullPrefix(SIGNER)
+                                .payingWith(SIGNER),
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount),
+                        getTxnRecord(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
```

Comment:
We need to get the record info for the call "SIGNER_MINTS_WITH_SIGNER_PUBLIC_KEY_AND_WRONG_CONTRACT_ID".
The "SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS" is for the previous ContractCall

Reply:
fixed